python/ray/serve/_private/application_state.py

```diff
@@ -309,35 +309,10 @@ def apply_deployment_args(
             RayServeException: If there is more than one deployment with
                 a non-null route prefix or docs path.
         """
-        # Makes sure that at most one deployment has a non-null route
-        # prefix and docs path.
-        num_route_prefixes = 0
-        num_docs_paths = 0
-        for deploy_param in deployment_params:
-            if deploy_param.get("route_prefix") is not None:
-                self._route_prefix = deploy_param["route_prefix"]
-                num_route_prefixes += 1
-
-            if deploy_param.get("docs_path") is not None:
-                self._docs_path = deploy_param["docs_path"]
-                num_docs_paths += 1
-        if num_route_prefixes > 1:
-            raise RayServeException(
-                f'Found multiple route prefixes from application "{self._name}",'
-                " Please specify only one route prefix for the application "
-                "to avoid this issue."
-            )
-        # NOTE(zcin) This will not catch multiple FastAPI deployments in the application
-        # if user sets the docs path to None in their FastAPI app.
-        if num_docs_paths > 1:
-            raise RayServeException(
-                f'Found multiple deployments in application "{self._name}" that have '
-                "a docs path. This may be due to using multiple FastAPI deployments "
-                "in your application. Please only include one deployment with a docs "
-                "path in your application to avoid this issue."
-            )
```

Comment:
I've moved around the code in this block to fit the new refactored architecture.